include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -854,9 +858,17 @@ struct OrtApi {
   /**
      * \param value A tensor created from OrtCreateTensor... function.
      * \param s A null terminated string.
-     * \param index index of string tensor element to fill 
+     * \param index index of string tensor element to fill
      */
   ORT_API2_STATUS(FillStringTensorElement, _Inout_ OrtValue* value, _In_ const char* s, size_t index);
+
+  /**
+     * Set a single session configuration entry as a pair of an enum and a string
+     * \param config_key An enum as the key of the configuration
+     * \param config_value A null terminated string representation of the config value
+     */
+  ORT_API2_STATUS(SetSessionConfiguration, _Inout_ OrtSessionOptions* options,
```

Comment:
Should we call this AddSessionOption()? With the enum change from OrtSessionConfigKey -> OrtSessionOptionKey, this would be clearer. And config_value can be option_value.